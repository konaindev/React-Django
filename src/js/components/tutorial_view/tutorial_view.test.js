import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";

import TutorialView from "./index";

jest.mock("react-responsive-modal", () => "Modal");
const _ = x => createStore(() => ({ tutorial: { tutorialView: x } }));

describe("TutorialView", () => {
  it("render default", () => {
    const props = {
      static_url: "static/",
      is_show_tutorial: true
    };
    const tree = renderer
      .create(
        <Provider store={_(props)}>
          <TutorialView />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("render hidden", () => {
    const props = {
      static_url: "static/",
      is_show_tutorial: false
    };
    const tree = renderer
      .create(
        <Provider store={_(props)}>
          <TutorialView />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
