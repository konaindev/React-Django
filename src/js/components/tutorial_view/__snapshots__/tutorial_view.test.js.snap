// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialView render default 1`] = `
<Modal
  center={true}
  classNames={
    Object {
      "closeButton": "modal-window__close",
      "modal": "modal-window tutorial-modal",
      "overlay": "modal-overlay",
    }
  }
  closeOnOverlayClick={true}
  focusTrapped={false}
  onClose={[Function]}
  open={true}
  styles={
    Object {
      "closeIcon": Object {
        "display": "none",
      },
    }
  }
>
  <div
    className="modal-window__head tutorial-modal__header"
  >
    Quickstart
  </div>
  <div
    className="modal-window__body"
  >
    <div
      className="tutorial-modal__image"
      style={
        Object {
          "backgroundImage": "url(static/test-file-stub)",
        }
      }
    />
    <div
      className="tutorial-modal__caption"
    >
      Keep tabs on all of your properties and how they’re performing towards their goals.
    </div>
    <div
      className="tutorial-modal__nav"
    >
      <div>
        <div
          className="tutorial-modal__nav-indicator tutorial-modal__nav-indicator--active"
          data-index={0}
          onClick={[Function]}
        />
        <div
          className="tutorial-modal__nav-indicator"
          data-index={1}
          onClick={[Function]}
        />
        <div
          className="tutorial-modal__nav-indicator"
          data-index={2}
          onClick={[Function]}
        />
      </div>
      <button
        className="button button--highlight tutorial-modal__nav-button"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</Modal>
`;

exports[`TutorialView render hidden 1`] = `
<Modal
  center={true}
  classNames={
    Object {
      "closeButton": "modal-window__close",
      "modal": "modal-window tutorial-modal",
      "overlay": "modal-overlay",
    }
  }
  closeOnOverlayClick={true}
  focusTrapped={false}
  onClose={[Function]}
  open={false}
  styles={
    Object {
      "closeIcon": Object {
        "display": "none",
      },
    }
  }
>
  <div
    className="modal-window__head tutorial-modal__header"
  >
    Quickstart
  </div>
  <div
    className="modal-window__body"
  >
    <div
      className="tutorial-modal__image"
      style={
        Object {
          "backgroundImage": "url(static/test-file-stub)",
        }
      }
    />
    <div
      className="tutorial-modal__caption"
    >
      Keep tabs on all of your properties and how they’re performing towards their goals.
    </div>
    <div
      className="tutorial-modal__nav"
    >
      <div>
        <div
          className="tutorial-modal__nav-indicator tutorial-modal__nav-indicator--active"
          data-index={0}
          onClick={[Function]}
        />
        <div
          className="tutorial-modal__nav-indicator"
          data-index={1}
          onClick={[Function]}
        />
        <div
          className="tutorial-modal__nav-indicator"
          data-index={2}
          onClick={[Function]}
        />
      </div>
      <button
        className="button button--highlight tutorial-modal__nav-button"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</Modal>
`;
